avocado/core/dispatcher.py

```diff
@@ -196,3 +197,61 @@ def map_method(self, method_name, *args):
             except:
                 self.log.error('Error running method "%s" of plugin "%s": %s',
                                method_name, ext.name, sys.exc_info()[1])
+
+
+class VarianterDispatcher(Dispatcher):
+
+    def __init__(self):
+        super(VarianterDispatcher, self).__init__('avocado.plugins.varianter')
+
+    def __getstate__(self):
+        """
+        Very fragile pickle which works when all Varianter plugins are
+        available on both machines.
+
+        TODO: Replace this with per-plugin-refresh-mechanism
+        """
+        return {"extensions": getattr(self, "extensions")}
+
+    def __setstate__(self, state):
+        """
+        Very fragile pickle which works when all Varianter plugins are
+        available on both machines.
+
+        TODO: Replace this with per-plugin-refresh-mechanism
+        """
+        self.__init__()
+        self.extensions = state.get("extensions")
+
+    def _map_method(self, method_name, deepcopy=False, *args, **kwargs):
+        """
+        :warning: **kwargs are not supported for deepcopy=True
+        """
+        ret = []
+        for ext in self.extensions:
+            try:
+                if hasattr(ext.obj, method_name):
+                    method = getattr(ext.obj, method_name)
+                    if deepcopy:
+                        copied_args = [copy.deepcopy(arg) for arg in args]
+                        ret.append(method(*copied_args))
```

Comment:
because the non-copy method would be "slower" and the deepcopy version would potentially pass uncopied key-word arguments...

I prefer the current solution, but if you insist I can use your suggested version only with `raise NotImplemented` when `kwargs` are supplied to `deepcopy` version.